python/graphstorm/sagemaker/sagemaker_partition.py

```diff
@@ -0,0 +1,371 @@
+
+    sm_env = json.loads(os.environ['SM_TRAINING_ENV'])
+    hosts = sm_env['hosts']
+    current_host = sm_env['current_host']
+    world_size = len(hosts)
+    os.environ['WORLD_SIZE'] = str(world_size)
+    host_rank = hosts.index(current_host)
+
+    # NOTE: Ensure no logging has been done before setting logging configuration
+    logging.basicConfig(
+        level=getattr(logging, job_config.log_level.upper(), None),
+        format=f'{current_host}: %(asctime)s - %(levelname)s - %(message)s'
+        )
+
+    boto_session = boto3.session.Session(region_name=os.environ['AWS_REGION'])
+    sagemaker_session = sagemaker.session.Session(boto_session=boto_session)
+
+    if host_rank == 0:
+        for host in hosts:
+            logging.info("The %s IP is %s",
+                host, socket.gethostbyname(host))
+
+
+    for key, val in os.environ.items():
+        logging.debug("%s: %s", key, val)
+
+    leader_addr = os.environ['MASTER_ADDR']
+    # sync with all instances in the cluster
+    if host_rank == 0:
+        # sync with workers
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.bind((leader_addr, 12000))
+        sock.listen(world_size)
+
+        client_list = [None] * world_size
+        for i in range(1, world_size):
+            client, _ = sock.accept()
+            client_list[i] = client
+    else:
+        # sync with master
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        for _ in range(30):
+            try:
+                sock.connect((leader_addr, 12000))
+                break
+            except: # pylint: disable=bare-except
+                logging.debug("Rank %s trying to connect to leader %s",
+                    host_rank, leader_addr)
+                time.sleep(10)
+        logging.debug("Connected")
+        client_list = None
+
+    # write ip list info to disk
+    ip_list = []
+    ip_list_path = os.path.join(tmp_data_path, 'ip_list.txt')
+    with open(ip_list_path, 'w', encoding='utf-8') as f:
+        for host in hosts:
+            f.write(f"{socket.gethostbyname(host)}\n")
+            ip_list.append(socket.gethostbyname(host))
+
+    graph_data_s3_no_trailing = graph_data_s3[:-1] if graph_data_s3.endswith('/') else graph_data_s3
+    graph_config_s3_path = f"{graph_data_s3_no_trailing}/{metadata_filename}"
+    meta_info_file = os.path.join(tmp_data_path, metadata_filename)
+
+    if not os.path.exists(meta_info_file):
+        if host_rank == 0:
+            logging.debug("Downloading metadata file from %s into %s",
+                graph_config_s3_path, meta_info_file)
+        download_data_from_s3(graph_config_s3_path, tmp_data_path,
+            sagemaker_session=sagemaker_session)
+
+    with open(meta_info_file, 'r') as f: # pylint: disable=unspecified-encoding
+        graph_config = json.load(f)
+
+    logging.info("Downloading graph data from %s into %s",
+        graph_data_s3, tmp_data_path)
+    graph_data_path = download_graph(
```

Comment:
if skip_partitioning, do we need to download the graph data?

Reply:
Yes, the local data are used by data_dispatch to create the DistDGL files right?